In [6]:
import pandas as pd
df = pd.read_csv("heart.csv")
df

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  \
0     63    1   3       145   233    1        0      150      0      2.3   
1     37    1   2       130   250    0        1      187      0      3.5   
2     41    0   1       130   204    0        0      172      0      1.4   
3     56    1   1       120   236    0        1      178      0      0.8   
4     57    0   0       120   354    0        1      163      1      0.6   
..   ...  ...  ..       ...   ...  ...      ...      ...    ...      ...   
298   57    0   0       140   241    0        1      123      1      0.2   
299   45    1   3       110   264    0        1      132      0      1.2   
300   68    1   0       144   193    1        1      141      0      3.4   
301   57    1   0       130   131    0        1      115      1      1.2   
302   57    0   1       130   236    0        0      174      0      0.0   

     slope  ca  thal  target  
0        0   0     1       1  
1        0   0     2       1  
2        2   0     2       1  
3        2   0     2       1  
4        2   0     2       1  
..     ...  ..   ...     ...  
298      1   0     3       0  
299      1   0     3       0  
300      1   2     3       0  
301      1   1     3       0  
302      1   1     2       0  

[303 rows x 14 columns]

In [8]:
mean_value = df['trestbps'].mean()
print("Mean:", mean_value)
median_value = df['trestbps'].median()
print("Median:", median_value)
mode_value = df['trestbps'].mode()
print("Mode:", mode_value)
std_value = df['trestbps'].std()
print("Standard Deviation:", std_value)
var_value = df['trestbps'].var()
print("Variation:", var_value)
range_value = df['trestbps'].max() - df['trestbps'].min()
print("Range of Values:", range_value)

Mean: 131.62376237623764
Median: 130.0
Mode: 0    120
Name: trestbps, dtype: int64
Standard Deviation: 17.53814281351709
Variation: 307.5864533473212
Range of Values: 106


In [11]:
corr_matrix = df.corr()
print(corr_matrix["target"].sort_values(ascending=False))

target      1.000000
cp          0.433798
thalach     0.421741
slope       0.345877
restecg     0.137230
fbs        -0.028046
chol       -0.085239
trestbps   -0.144931
age        -0.225439
sex        -0.280937
thal       -0.344029
ca         -0.391724
oldpeak    -0.430696
exang      -0.436757
Name: target, dtype: float64


## Features:

- age: The person’s age in years
- sex: The person’s sex (1 = male, 0 = female)
- cp: chest pain type— Value 0: asymptomatic— Value 1: atypical angina— Value 2: non-anginal pain— Value 3: typical angina
- trestbps: The person’s resting blood pressure (mm Hg on admission to the hospital)
- chol: The person’s cholesterol measurement in mg/dl
- fbs: The person’s fasting blood sugar (> 120 mg/dl, 1 = true; 0 = false)
- restecg: resting electrocardiographic results— Value 0: showing probable or definite left ventricular hypertrophy by Estes’ criteria— Value 1: normal— Value 2: having ST-T wave abnormality (T wave inversions and/or ST elevation or depression of > 0.05 mV)
- thalach: The person’s maximum heart rate achieved
- exang: Exercise induced angina (1 = yes; 0 = no)
- oldpeak: ST depression induced by exercise relative to rest (‘ST’ relates to positions on the ECG plot. See more here)
- slope: the slope of the peak exercise ST segment — 0: downsloping; 1: flat; 2: upsloping0: downsloping; 1: flat; 2: upsloping
- ca: The number of major vessels (0–3)
- thal: A blood disorder called thalassemia Value 0: NULL (dropped from the dataset previouslyValue 1: fixed defect (no blood flow in some part of the heart)Value 2: normal blood flowValue 3: reversible defect (a blood flow is observed but it is not normal)
- target: Heart disease (1 = no, 0= yes)
